src/main/java/nl/tudelft/goalkeeper/parser/results/files/module/parsers/RuleParser.java

```diff
@@ -43,6 +45,9 @@ public static Rule parse(languageTools.program.agent.rules.Rule r) {
                     startingLine = Math.min(startingLine, source.getLine());
                     endingLine = Math.max(endingLine, source.getLine());
                 }
+                if (c.getKRLanguage() != KRLanguage.UNKNOWN) {
```

Comment:
Instead of performing this if statement could you not just do the assignment straight away? This would probably be faster than performing an if statement and then assigning the language (which is most likely case I think)

Reply:
No, because it's possible that a condition has an unknown KR language, e.g.: `if true then ...`. Which would mean that `if bel(x(A)), true then ...` would have a KR language of `UNKNOWN` while it's a `PROLOG` condition at the start. Although I must agree that it's not very efficient.

Reply:
Aaah fair enough. I did not think that `true` would be of type `UNKNOWN`